src/main.rs

```diff
@@ -140,6 +162,187 @@ async fn main() -> Result<()> {
+                                    .short("l")
+                                    .long("list")
+                                    .help("List devices and playlists"))
+                               .arg(Arg::with_name("devices")
+                                    .short("d")
+                                    .long("devices")
+                                    .help("List the category 'devices'"))
+                               .arg(Arg::with_name("playlists")
+                                    .short("p")
+                                    .long("playlists")
+                                    .help("List / search the category 'playlists'"))
+                               .arg(Arg::with_name("liked")
+                                    .long("liked")
+                                    .help("List the category 'liked songs'"))
+                               .group(ArgGroup::with_name("listable")
+                                    .args(&["devices", "playlists", "liked"])
+                                    .multiple(false))
+                               // Searching
+                               .arg(Arg::with_name("search")
+                                    .short("s")
+                                    .long("search")
+                                    .takes_value(true)
+                                    .value_name("SEARCH")
+                                    .requires("searchable")
+                                    .help("Search for tracks etc."))
+                               .arg(Arg::with_name("albums")
+                                    .short("b")
+                                    .long("albums")
+                                    .help("Search the category 'albums'"))
+                               .arg(Arg::with_name("artists")
+                                    .short("a")
+                                    .long("artists")
+                                    .help("Search the category 'artists'"))
+                               .arg(Arg::with_name("tracks")
+                                    .short("t")
+                                    .long("tracks")
+                                    .help("Search the category 'tracks'"))
+                               .arg(Arg::with_name("shows")
+                                    .short("w")
+                                    .long("shows")
+                                    .help("Search the category 'shows'"))
+                               .group(ArgGroup::with_name("searchable")
+                                    .args(&["playlists", "tracks", "albums", "artists", "shows"])
+                                    .multiple(false))
+                               // Actions
+                               .group(ArgGroup::with_name("actions")
+                                    .args(&["list", "search"])
+                                    .multiple(false)
+                                    .required(true)))
```

Comment:
Would you be able to move this clap construct into in the cli module please? Just because it's so large 😄.

Could create a `cli` dir with `cli.rs` and `clap.rs` or something like that. We could use this module later to further breakdown the `cli.rs` file.

Reply:
Also, I've been meaning to try out [structopt](https://github.com/TeXitoi/structopt), which might simplify this.

But don't worry about that for this PR.

Reply:
Makes sense, it's pretty big.